### PR TITLE
Add Exceptions basic doc

### PR DESCRIPTION
**Pull request by lmazuel**
Adding in documentation a detailled overview of the possible exceptions while using Autorest generated code.

Preview is available on: http://laurent-python-sdk.readthedocs.io/en/exceptions_doc/exceptions.html

### CODE REVIEW

File: doc/exceptions.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Exceptions handling
```
Comment (huguesv) [inline doc/exceptions.rst]: Exception handling (not plural)

File: doc/exceptions.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+With no sub-classing from the service team, the basic exception you can expect from an operation is :exc:`CloudError<msrestazure:msrestazure.azure_exceptions.CloudError>`. 
+In some situation however, the exceptions are specialized by package (see `azure-batch` package use :exc:`BatchErrorException<azure.batch.models.BatchErrorException>`).
+If a specialized exception is used, it will always be a sub-class at some level of :exc:`ClientException<msrest.exceptions.ClientException>`.
+
+Either you should retry or not the same query for these exceptions depends on the service and the error content and cannot be generalized on this article.
+For instance, an update to a SQL Database might be refused because you have another underlying operation (in this case you can retry), or a create can be refused
+because you don't respect the acceptable pattern name (in this case retry will never work). Use the `error` and `response` attribute of
+:exc:`CloudError<msrestazure:msrestazure.azure_exceptions.CloudError>` to decide.
+
+All these operations can also send this set of generic exceptions that are defined in the :mod:`msrest.exceptions<msrest:msrest.exceptions>` module:
```
Comment (huguesv) [inline doc/exceptions.rst]: send -> raise

File: doc/exceptions.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Every exception related to the service content will be indicated in the docstring of the method. For instance, for this 
+:meth:`create_or_update<azure.mgmt.network.operations.NetworkSecurityGroupsOperations.create_or_update>` operation
+you can see that you might receive a :exc:`CloudError<msrestazure:msrestazure.azure_exceptions.CloudError>`.
+
+With no sub-classing from the service team, the basic exception you can expect from an operation is :exc:`CloudError<msrestazure:msrestazure.azure_exceptions.CloudError>`. 
+In some situation however, the exceptions are specialized by package (see `azure-batch` package use :exc:`BatchErrorException<azure.batch.models.BatchErrorException>`).
+If a specialized exception is used, it will always be a sub-class at some level of :exc:`ClientException<msrest.exceptions.ClientException>`.
+
+Either you should retry or not the same query for these exceptions depends on the service and the error content and cannot be generalized on this article.
```
Comment (huguesv) [inline doc/exceptions.rst]: Either -> Whether

File: doc/exceptions.rst [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  - :exc:`AuthenticationError<msrest.exceptions.AuthenticationError>`: password invalid/expired, etc. Retry will likely not work.
+  - :exc:`ClientRequestError<msrest.exceptions.ClientRequestError>`: `requests` library raised an exception (connection error, mostly HTTP level issues). Retry may work (if it's because your base_url is not correct, might not work on retry)
+  - :exc:`DeserializationError<msrest.exceptions.DeserializationError>`: unexpected RestAPI answer. Retry will likely not work. Please create an issue on Github if you see this exception.
+  - :exc:`HttpOperationError<msrest.exceptions.HttpOperationError>`: bad HTTP status code from Azure. See inner exception to decide if it's worth retrying.
+  - :exc:`SerializationError<msrest.exceptions.SerializationError>`: unable to serialize your request. Your Python code didn't respect the expected model. Retry will never work. Please create an issue on Github if you see this exception and are 100% your parameters are correct.
+  - :exc:`TokenExpiredError<msrest.exceptions.TokenExpiredError>`: please renew your credentials. Likely retry with new credentials will be ok.
```
Comment (huguesv) [inline doc/exceptions.rst]: Is this documented somewhere (how to renew credentials)?

File: doc/exceptions.rst [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  - :exc:`ValidationError<msrest.exceptions.ValidationError>`: dynamic check of your request failed. Fix your parameters with expected value. For instance, you didn't respect the regexp for the account name. This is static checking, will never work on retry.
```
Comment (huguesv) [inline doc/exceptions.rst]: this both says that it's a dynamic check and a static check.

this is the client side parameter validation, ie. not server side.

**Comment by huguesv** [comment]
Are there special considerations for exception handling when using long running operations?  Also when using `raw=true`?

File: doc/exceptions.rst (hunk 4 above)
Comment (lmazuel) [inline doc/exceptions.rst]: That is a good question. @annatisch what do you think a user should do if he gets this exception?

**Comment by lmazuel** [comment]
@huguesv added chapter for async and raw if you can re-read again :)

File: doc/exceptions.rst [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+and the second one might fail during polling the status of the operation
+
+.. important:: Old version of the packages never failed on the first call, but this behavior was replaced by the one described and you should follow
+               this pattern even for old packages.
+
+Raw operation
++++++++++++++
+
+All operation accept a `raw=True` parameter to indicate that the method must return the `requests.Response` instance directly.
+All the previous exceptions are still corect, but :exc:`DeserializationError<msrest.exceptions.DeserializationError>`, since we do not deserialize the answer.
```
Comment (huguesv) [inline doc/exceptions.rst]: corect -> correct

but -> except for

**Pull request merged**